File: frontend-appstore/PusulaService/Models/AdminModels.swift
import Foundation

// MARK: - Dashboard KPIs

struct DashboardKPIs: Codable {
    let monthlyRevenue: Double?
    let outstandingDebt: Double?
    let netProfit: Double?
    let profitMargin: Double?
    let activeTickets: Int?
    let completedThisMonth: Int?
    let inventoryValue: Double?
}

// MARK: - Technician Stats

struct TechnicianStat: Codable, Identifiable {
    let userId: Int
    let fullName: String?
    let completedToday: Int?
    let completedThisMonth: Int?
    let collectedToday: Double?
    let collectedThisMonth: Double?
    let activeTickets: Int?
    let lastLocation: String? // "lat,lon"
    
    var id: Int { userId }
}

// MARK: - Profit Analysis

struct ProfitAnalysis: Codable {
    let totalCostOfGoodsSold: Double?
    let totalRevenueFromParts: Double?
    let grossProfit: Double?
    let grossMarginPercent: Double?
    let topProfitableParts: [PartProfit]?
}

struct PartProfit: Codable, Identifiable {
    let partName: String
    let buyPrice: Double?
    let sellPrice: Double?
    let quantitySold: Int?
    let totalProfit: Double?
    let marginPercent: Double?
    
    var id: String { partName }
}

// MARK: - Quota Status

struct QuotaStatus: Codable {
    let planName: String?
    let quotas: [QuotaItem]?
}

struct QuotaItem: Codable, Identifiable {
    let featureKey: String
    let featureLabel: String?
    let currentUsage: Int?
    let limit: Int?
    let usagePercent: Double?
    
    var id: String { featureKey }
}

// MARK: - Field Radar Pin

struct FieldPin: Codable, Identifiable {
    let technicianId: Int
    let technicianName: String?
    let coordinates: String? // "lat,lon"
    let customerName: String?
    let ticketStatus: String?
    let ticketId: Int?
    
    var id: Int { technicianId }
    
    var latitude: Double? {
        guard let coords = coordinates?.split(separator: ","), coords.count == 2 else { return nil }
        return Double(coords[0])
    }
    
    var longitude: Double? {
        guard let coords = coordinates?.split(separator: ","), coords.count == 2 else { return nil }
        return Double(coords[1])
    }
}

// MARK: - Admin Service

enum AdminService {
    static func getDashboardKPIs() async throws -> DashboardKPIs {
        try await NetworkManager.shared.get("/api/admin/dashboard")
    }
    
    static func getTechnicianStats() async throws -> [TechnicianStat] {
        try await NetworkManager.shared.get("/api/admin/technician-stats")
    }
    
    static func getProfitAnalysis() async throws -> ProfitAnalysis {
        try await NetworkManager.shared.get("/api/admin/profit-analysis")
    }
    
    static func getQuotaStatus() async throws -> QuotaStatus {
        try await NetworkManager.shared.get("/api/admin/quota-status")
    }
    
    static func getFieldRadar() async throws -> [FieldPin] {
        try await NetworkManager.shared.get("/api/admin/field-radar")
    }
}

File: frontend-playstore/PusulaService/app/src/main/java/com/pusula/service/data/model/AdminModels.kt
package com.pusula.service.data.model

data class DashboardKPIs(
    val monthlyRevenue: Double? = null,
    val outstandingDebt: Double? = null,
    val netProfit: Double? = null,
    val profitMargin: Double? = null,
    val activeTickets: Int? = null,
    val completedThisMonth: Int? = null,
    val inventoryValue: Double? = null
)

data class TechnicianStat(
    val userId: Long,
    val fullName: String? = null,
    val completedToday: Int? = null,
    val completedThisMonth: Int? = null,
    val collectedToday: Double? = null,
    val collectedThisMonth: Double? = null,
    val activeTickets: Int? = null,
    val lastLocation: String? = null
)

data class ProfitAnalysis(
    val totalCostOfGoodsSold: Double? = null,
    val totalRevenueFromParts: Double? = null,
    val grossProfit: Double? = null,
    val grossMarginPercent: Double? = null,
    val topProfitableParts: List<PartProfit>? = null
)

data class PartProfit(
    val partName: String,
    val buyPrice: Double? = null,
    val sellPrice: Double? = null,
    val quantitySold: Int? = null,
    val totalProfit: Double? = null,
    val marginPercent: Double? = null
)

data class QuotaStatus(
    val planName: String? = null,
    val quotas: List<QuotaItem>? = null
)

data class QuotaItem(
    val featureKey: String,
    val featureLabel: String? = null,
    val currentUsage: Int? = null,
    val limit: Int? = null,
    val usagePercent: Double? = null
)

data class FieldPin(
    val technicianId: Long,
    val technicianName: String? = null,
    val coordinates: String? = null,
    val customerName: String? = null,
    val ticketStatus: String? = null,
    val ticketId: Long? = null
)
